lms/djangoapps/class_dashboard/dashboard_data.py

```diff
@@ -24,6 +26,25 @@
     'lti',
 ]
 
+# exclude these in Metrics
+NON_STUDENT_ROLES = ['instructor', 'staff']
+
+
+def get_non_student_list(course_key):
+    """
+    Find all user_ids with instructor or staff roles in student_courseaccessrole table
+    """
+
+    non_student_idlist = []
+    non_students = CourseAccessRole.objects.select_related('user').filter(
+        course_id=course_key,
+        role__in=NON_STUDENT_ROLES,
+    ).values('user__id')
+
+    non_student_idlist = [non_student['user__id'] for non_student in non_students]
+
+    return non_student_idlist
```

Comment:
You could also do return [non_student['user__id'] for non_student in non_students] - I think that's a little bit more pythonic.